src/elementDisabler.ts

```diff
@@ -80,6 +80,12 @@ function disableFormElements (form: HTMLFormElement): void {
 function disableFormElement (element: HTMLFormElement): void {
   if (element.dataset.ujsDisabled != null) return
 
+  const form = element?.form
+
+  // If the form is using the browser's default constraint validation, and the form is not valid, we should
+  // not attempt to disable the button because the browser will prevent errors and prevent submission
+  if (form?.noValidate == false && form?.checkValidity() == false) { return }
```

Comment:
Seems like we should match return style above:

```suggestion
  if (form?.noValidate == false && form?.checkValidity() == false) return
```